README.md

```diff
@@ -200,17 +197,10 @@ You can execute all test programs through cmake and ctest with:
 
     Total Test time (real) =   1.37 sec
 
```

Comment:
That's the part that is the most important to me. We can document both options to use this library, the find_package and the the FetchContent or ExternalProject....

Reply:
Surely every single person who would want to use it, already knows how to use it without giving generic cmake how-to guides in a highly specific project README?

Reply:
For `find_package()`, it's nice to show it in the README, because it's convenient for users to quickly see the `find_package()` call convention for the library and the imported target, without digging into the install tree (or CMakeLists) to check whether upstram provides a config file or not.

For `FetchContent()`, as you wish, but if only one should be kept, it should be find_package() which is by far the most robust approach for dependency management. And for sure only the ALIAS or IMPORTED target should be displayed in the README.
We could also document `add_subdirectory()` etc, but as @eli-schwartz said, it's too much.

Reply:
Ok I tested and it works fine with FetchContent.

https://gist.github.com/bsergean/1cb658e6280a578af5bef1c26d9f7dd3

The one thing I think we should do is have an option to build the command/test programs, and default it to false, but maybe the author of the library likes them to be built all the time.

Reply:
When you build the library current you also build the test programs which is a bit unfortunate. It can make build fails if some of the test programs have warnings for example, but the library does not, and the main consumer builds everything with warning as errors or similar.

```
[ 73%] Built target my_exe
[ 78%] Building C object _deps/libdeflate-build/programs/CMakeFiles/libdeflate_programs_common.dir/prog_util.c.o
[ 84%] Building C object _deps/libdeflate-build/programs/CMakeFiles/libdeflate_programs_common.dir/tgetopt.c.o
[ 89%] Linking C static library liblibdeflate_programs_common.a
[ 89%] Built target libdeflate_programs_common
[ 94%] Building C object _deps/libdeflate-build/programs/CMakeFiles/libdeflate-gzip.dir/gzip.c.o
[100%] Linking C executable libdeflate-gzip
```

Reply:
The cmake && make && ctest guidance can be rationalized as instructions for people who don't really know much about cmake and just want to know how to build the project. The alternative is linking to the cmake docs, and expecting them to read that in order to get the basic "I just want to install this interesting software, help!" working.

And indeed, mentioning the correct name for use in find_package() is useful in ways that don't constitute "don't repeat yourself, cmake has developer docs already". Because the most important info there is "what's the package config name?"

Starting to discuss FetchContent() is where this crosses over into debatably useless information no one cares about, because most people won't use it and the few people who do use it already know it.

It's fundamentally a feature of someone else's cmake, not the libdeflate cmake, and the only project specific part of FetchContent() is the same guidance that was offered for find_package. So discussing find_package encompasses both, while reducing the cognitive overhead of the README.

Reply:
@bsergean regarding utility, yes it's intended because it's not a test. An option could be added to disable it. But the Makefile builts & install it.